pkg/storage/storage.go

```diff
@@ -251,13 +251,13 @@ func makeKey(rlsname string, version int) string {
 
 // Init initializes a new storage backend with the driver d.
 // If d is nil, the default in-memory driver is used.
-func Init(d driver.Driver) *Storage {
+func Init(d driver.Driver, log func(format string, args ...interface{})) *Storage {
```

Comment:
As a note, this would be considered a breaking change to the function signature. As per our [semver contract](https://github.com/helm/helm/blob/master/CONTRIBUTING.md#semantic-versioning), we have committed to not making any changes to existing function signatures until Helm 4 as much as possible, though we have made exceptions in the past to fix regressions or security flaws. Given this is a small quality-of-life change I'm not sure this warrants breaking backwards compatibility.

Have you considered alternative solutions to address this issue?

Reply:
Yep, this change might go to be a break change. I have change this to add a `NewStorage` function, this will be  backwards compatibility.